### ISSUE TITLE
[BUG] WhatsApp/Media/*/ containing no images or folders

### ISSUE THREAD

**Issue by Beastelson** [T0biasCZe/AdbFileManager#25]
Any subfolder of WhatsApp/Media/ is showing as containing no files when clicked, though there are clearly files present.

![image](https://github.com/T0biasCZe/AdbFileManager/assets/29049636/d1cb7617-b125-46d2-b8a0-af4e3acd3425)

No errors are thrown, and when copying WhatsApp Images (the entire folder) from the WhatsApp/Media/ folder, the folder copied as expected, including all images and subfolders. 
This holds for any of the folders present in the WhatsApp/Media/ folder.
 
Using android 9 on Motorola Moto g6 plus.

**Comment by T0biasCZe** [comment]
in the bottom right corner, there is a small button that will show debug console
![image](https://github.com/T0biasCZe/AdbFileManager/assets/44525446/95c8ff4d-e1b5-4f21-a105-e4021450f599)

could you try looking into there if it prints the contents of the whatsapp folder in there?

**Comment by T0biasCZe** [comment]
In the debug console it should look like this:
![image](https://github.com/T0biasCZe/AdbFileManager/assets/44525446/0b203cc4-f8a8-41f2-8ed3-f85d0c9e052f)

I cant unfortunately test it myself because whatsapp doesnt use that folder anymore on android 11+ phones

**Comment by ChrisRahme** [comment]
I have the same problem, here's the relevant debug part:

```
adb shell ls "/sdcard/Android/media/com.whatsapp/WhatsApp/Media/"
WallPaper
WhatsApp Images
WhatsApp Stickers
WhatsApp Video
WhatsApp Voice Notes

fastchecking if "/sdcard/Android/media/com.whatsapp/WhatsApp/Media//WhatsApp Images" is a folder: False
fastchecking if "/sdcard/Android/media/com.whatsapp/WhatsApp/Media//WhatsApp Stickers" is a folder: False
fastchecking if "/sdcard/Android/media/com.whatsapp/WhatsApp/Media//WhatsApp Video" is a folder: False
fastchecking if "/sdcard/Android/media/com.whatsapp/WhatsApp/Media//WhatsApp Voice Notes" is a folder: False
```

**Comment by T0biasCZe** [comment]
and in the list it shows the "no files found"?

**Comment by ChrisRahme** [comment]
Yup

**Comment by Beastelson** [comment]
Sorry for not getting back earlier!

Compatibility Disabled:
`CellMouseDoubleClick()
cur_path_TextChanged();
cur_path_TextChanged false
adb shell ls -lL "/sdcard/WhatsApp/Media/"
total 796
drwxrwx--x   2 root sdcard_rw   4096 2023-07-02 00:30 WallPaper
drwxrwx--x   4 root sdcard_rw  36864 2024-03-13 12:18 WhatsApp Animated Gifs
drwxrwx--x   4 root sdcard_rw   4096 2024-01-31 10:53 WhatsApp Audio
drwxrwx--x   4 root sdcard_rw  16384 2024-03-10 14:39 WhatsApp Documents
drwxrwx--x   4 root sdcard_rw 540672 2024-03-25 20:57 WhatsApp Images
drwxrwx--x   2 root sdcard_rw   4096 2023-07-02 00:30 WhatsApp Profile Photos
drwxrwx--x   2 root sdcard_rw 135168 2024-03-20 14:21 WhatsApp Stickers
drwxrwx--x   4 root sdcard_rw  12288 2024-03-25 11:51 WhatsApp Video
drwxrwx--x   3 root sdcard_rw   4096 2023-08-06 10:04 WhatsApp Video Notes
drwxrwx--x 132 root sdcard_rw   4096 2024-03-15 19:21 WhatsApp Voice Notes

CellMouseDoubleClick()
cur_path_TextChanged();
cur_path_TextChanged false
adb shell ls -lL "/sdcard/WhatsApp/Media/WhatsApp Images/"`

Compatibility Enabled:
`CellMouseDoubleClick()
checking if "d---------" is a folder: True
cur_path_TextChanged();
cur_path_TextChanged false
adb shell ls "/sdcard/WhatsApp/Media/"
WallPaper
WhatsApp Animated Gifs
WhatsApp Audio
WhatsApp Documents
WhatsApp Images
WhatsApp Profile Photos
WhatsApp Stickers
WhatsApp Video
WhatsApp Video Notes
WhatsApp Voice Notes

checking if "/sdcard/WhatsApp/Media//WhatsApp Animated Gifs" is a folder: True
checking if "/sdcard/WhatsApp/Media//WhatsApp Audio" is a folder: True
checking if "/sdcard/WhatsApp/Media//WhatsApp Documents" is a folder: True
checking if "/sdcard/WhatsApp/Media//WhatsApp Images" is a folder: True
checking if "/sdcard/WhatsApp/Media//WhatsApp Profile Photos" is a folder: True
checking if "/sdcard/WhatsApp/Media//WhatsApp Stickers" is a folder: True
checking if "/sdcard/WhatsApp/Media//WhatsApp Video" is a folder: True
checking if "/sdcard/WhatsApp/Media//WhatsApp Video Notes" is a folder: True
checking if "/sdcard/WhatsApp/Media//WhatsApp Voice Notes" is a folder: True
CellMouseDoubleClick()
checking if "d---------" is a folder: True
cur_path_TextChanged();
cur_path_TextChanged false
adb shell ls "/sdcard/WhatsApp/Media/WhatsApp Images/"`

Let me know if I can provide anything else!

**Comment by angelbla** [comment]
The bug seems to happen to me only when the path contains spaces. All folders "Whatsapp Images", "Whatsapp videos" etc contain a space. I've checked some other local paths and whenever it contains a space, if I access that folder ADBFileManager shows "no files" in it. Although there are.
Maybe it's related to some kind of space-escaping problem in the adb command?

**Comment by T0biasCZe** [comment]
yeah it was inproper space escaping
the directory was escaped with just "
![image](https://github.com/user-attachments/assets/edca94dd-c81e-47bc-b484-e8c1a3559f70)
adding ' fixed the issue
![image](https://github.com/user-attachments/assets/3416113d-7ce7-4e58-b767-7f3ee902a4bf)

**Comment by T0biasCZe** [comment]
fixed in v1.9
https://github.com/T0biasCZe/AdbFileManager/releases/tag/v1.9

**Comment by T0biasCZe** [comment]
Feel free to reopen the issue if the issue sometimes still occurs 
(I tested only on Android 4.4 with compatibility mode, however it should be fixed on newer android versions too)
